src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -963,9 +963,22 @@ int xcldev::xclTop(int argc, char *argv[])
 }
 
 const std::string dsaPath("/opt/xilinx/dsa/");
-const std::string xsaPath("/opt/xilinx/xsa/");
 const std::string xrtPath("/opt/xilinx/xrt/");
 
+inline std::string getXsaPath(const uint16_t &vendor)
+{
+    std::string vendorName;
+    switch (vendor) {
+        case ARISTA_ID:
+            vendorName = "arista";
+            break;
+        default:
+            vendorName = "xilinx";
+            break;
+    }
+    return vendor ? "/opt/" + vendorName + "/xsa/" : std::string();
```

Comment:
Assuming the variable "vendor" was check earlier, we know that it isn't zero, so this line now becomes:
`return  "/opt/" + vendorName + "/xsa/";`